internal/pkg/deploy/cloudformation/app.go

```diff
@@ -313,12 +373,50 @@ func (cf CloudFormation) RemoveJobFromApp(app *config.Application, jobName strin
 	return nil
 }
 
+// RemoveEnvFromAppOpts contains the parameters to call RemoveEnvFromApp.
+type RemoveEnvFromAppOpts struct {
+	App                  *config.Application
+	EnvName              string
+	EnvAccountID         string
+	EnvRegion            string
+	DeleteStackInstance  bool // If this is the last environment in a region, we can delete it.
+	RemoveAccountFromApp bool // If this is the last environment in an account, we need to redeploy the app stack.
+}
+
+// RemoveEnvFromApp optionally redeploys the app stack to remove the account and deletes the stackset
+// instance for that region. This method cannot check that deleting the stackset or removing the app
+// won't break copilot. Be careful.
+func (cf CloudFormation) RemoveEnvFromApp(opts *RemoveEnvFromAppOpts) error {
```

Comment:
Why don't we move the logic for determining the value for "DeleteStackInstance" and "RemoveAccountFromApp" here in this function? So as to make sure calling `RemoveEnvFromApp` is always safe.

Reply:
We'll have to duplicate this logic somewhere, though, because we have to check in env_delete.go whether or not to empty the s3 buckets and ecr repos. I don't love that, but I see the advantage of having logic in the RemoveEnvFromApp call.

Reply:
> because we have to check in env_delete.go whether or not to empty the s3 buckets and ecr repos.

Maybe we can let `RemoveEnvFromApp` decide if we need to empty the s3 buckets and ecr repo before deleting the stackset?